cmake/kokkos_enable_devices.cmake

```diff
@@ -26,6 +26,9 @@ IF(Kokkos_ENABLE_PTHREAD)  # for backward compatibility
   LIST(APPEND KOKKOS_ENABLED_DEVICES THREADS)
   SET(KOKKOS_HAS_HOST ON)
   MESSAGE(DEPRECATION "The Kokkos_ENABLE_PTHREAD option is deprecated. Use Kokkos_ENABLE_THREADS instead!")
+ELSEIF(NOT Kokkos_ENABLE_THREADS)  # for backward compatibility
+  SET(Kokkos_ENABLE_PTHREAD OFF CACHE BOOL "Whether to build C++ threads backend")
+  SET(KOKKOS_ENABLE_PTHREAD OFF)
```

Comment:
Please elaborate why this is needed

Reply:
This is to ensure that `{Kokkos,KOKKOS}_ENABLE_PTHREAD` has a default definition. Prior to the renaming to Threads, this was done by calling `KOKKOS_DEVICE_OPTION` with `PTHREAD`; currently this is done for `THREADS` (line 22 defaulted to OFF) with some follow-up if `Kokkos_ENABLE_PTHREAD` is already defined (lines 23-28), but there is no default "OFF" case for `Kokkos_ENABLE_PTHREAD`
This was needed with deprecated code enabled in some Trilinos packages (e.g. Tpetra) that check on the definition of `Kokkos_ENABLE_PTHREAD` during configuration